cmake/templates/python_win32_wrapper.cpp

```diff
@@ -0,0 +1,177 @@
+            {
+                // buffer not large enough
+                moduleName.resize(moduleName.size() * 2);
+                continue;
+            }
+            moduleName.resize(result);
+            break;
+        }
+        return moduleName;
+    };
+
+    const auto FindPythonScript = [](const std::string& exeName) -> std::string
+    {
+        // this could become more fail-proof and readable with https://en.cppreference.com/w/cpp/filesystem from c++20
+        const std::string exeExtension = ".exe";
+        if (exeName.size() <= exeExtension.size() || exeName.substr(exeName.size() - exeExtension.size()) != exeExtension)
+        {
+            throw L"Invalid name.";
+        }
+
+        // use quoted string to ensure the correct path is used
+        return "\"" + exeName.substr(0, exeName.size() - exeExtension.size()) + "\"";
+    };
+
+    const auto GetPythonExecutable = []() -> std::string
+    {
+        std::string pythonExecutable = "python";
+
+        // use quoted string to indicate where the file name ends and the arguments begin
+        return "\"" + pythonExecutable + "\"";
+    };
+
+    const auto pythonExecutable = GetPythonExecutable();
+    const auto pythonScript = FindPythonScript(GetCurrentModuleName());
+    std::string command = pythonExecutable + " " + pythonScript;
+    for (auto i = 1; i < argc; ++i)
+    {
+        command += " ";
+        // use quoted strings to handle spaces within each argument
+        command += " \"" + std::string(argv[i]) + "\"";
+    }
+
+#if defined(DEBUG)
+    for (auto i = 0; i < argc; ++i)
+    {
+        printf("[DEBUG] %d:\t%s\n", i, argv[i]);
+    }
+
+    printf("[DEBUG] script: %s\n", pythonScript.c_str());
+    printf("[DEBUG] command: %s\n", command.c_str());
+#endif
+
+    STARTUPINFO startup_info;
+    PROCESS_INFORMATION process_info;
+    ::memset(&startup_info, 0, sizeof(startup_info));
+    ::memset(&process_info, 0, sizeof(process_info));
+    startup_info.cb = sizeof(startup_info);
+
+    auto result = ::CreateProcess(
+        nullptr,                // program to execute (nullptr = execute command line)
+        &command[0],            // command line to execute
+        nullptr,                // process security attributes
+        nullptr,                // thread security attributes
+        false,                  // determines if handles from parent process are inherited
+        0,                      // no creation flags
+        nullptr,                // enviornment (nullptr = use parent's)
+        nullptr,                // current directory (nullptr = use parent's)
+        &startup_info,          // startup info
+        &process_info           // process info
+    );
+    if (!result)
+    {
+        const auto error = ::GetLastError();
+        switch (error)
+        {
+        case ERROR_FILE_NOT_FOUND:
+        {
+            std::cout << "Error! Python executable cannot be found, is it added to PATH?" << std::endl;
```

Comment:
Please use `printf` here to avoid interleaved output. Also error messages should go to `stderr`.

Same below.

Reply:
change to use `fprintf(stderr, ...)`